lib/rules/selector-max-specificity/index.cjs

```diff
@@ -104,6 +108,45 @@ const rule = (primary, secondaryOptions) => {
 				return selectorSpecificity.compare(childSpecificity, maxSpec) > 0 ? childSpecificity : maxSpec;
 			}, zeroSpecificity());
 
+		/**
+		 * If a `of <selector>` (`An+B of S`) is found in the specified pseudo node,
+		 * returns a copy of the pseudo node, ignoring a `of` selector (`An+B of S`).
+		 * Otherwise, returns the specified node as-is.
+		 *
+		 * @see https://drafts.csswg.org/selectors/#the-nth-child-pseudo
+		 * @param {import('postcss-selector-parser').Pseudo} pseudo
+		 * @returns {import('postcss-selector-parser').Pseudo}
+		 */
+		const ignoreOfSelectorIfAny = (pseudo) => {
+			/** @type {(node: import('postcss-selector-parser').Node | undefined) => boolean} */
+			const isOfSelector = (node) => Boolean(node && node.type === 'tag' && node.value === 'of');
+
+			/** @type {(node: import('postcss-selector-parser').Node | undefined) => boolean} */
+			const isSpace = (node) => Boolean(node && node.type === 'combinator' && node.value === ' ');
+
+			const nodes = pseudo.nodes[0]?.nodes ?? [];
+			const ofSelectorIndex = nodes.findIndex((child, i, children) => {
+				// Find ' of <selector>' nodes
+				return isSpace(child) && isOfSelector(children[i + 1]) && isSpace(children[i + 2]);
```

Comment:
minor nit

not sure if the `node.type ===` check are needed
which could allow a refactoring
e.g. `const foo = (node, value) => Boolean(node && && node.value === value);`